src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/IResiliencyReportData.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Diagnostics.ModelsAndUtils.Models.ResponseExtensions
+{
+    public interface IResiliencyReportData
+    {
+        /// <summary>
+        /// Customer's name used for the report's cover. This will normally be either customer's Company or simply Customer's name obtained from the subscription.
+        /// </summary>
+        string CustomerName { get; set; }
+
+        /// <summary>
+        /// Array containing a list of all the resources to be included in the report along with their 
+        /// </summary>
+        ResiliencyResource[] GetResiliencyResourceList();
+
+        /// <summary>
+        /// Array containing a list of all the resources to be included in the report along with their 
+        /// </summary>
+        void SetResiliencyResourceList(ResiliencyResource[] value);
+
+             
+    }
+
+    /// <summary>
+    /// ResiliencyResources supported can be Web App or ASE
+    /// </summary>
+    public interface IResiliencyResource
+    {
+        /// <summary>
+        /// Resource name
+        /// </summary>
+
+        string Name { get; set; }
+        /// <summary>
+        /// Overall Score for the Resource is calculated based on the grade of each feature
+        /// </summary>
+        double CalculateOverallScore();
+
+        double OverallScore { get; }
+
+
+        /// <summary>
+        /// Retrieve the list of Resiliency Features
+        /// </summary>
+        ResiliencyFeature[] GetResiliencyFeaturesList();
+
+
+        /// <summary>
+        /// Set the list of Resiliency Features
+        /// </summary>
+        void SetResiliencyFeaturesList(ResiliencyFeature[] value);
```

Comment:
You can sum this up like so:

```suggestion
        /// <summary>
        /// Retrieve the list of Resiliency Features
        /// </summary>
        public ResiliencyFeature[] FeaturesList {get; set;}
```

Reply:
That's what I did initially but when I do I get the following:
CA1819: Properties should not return arrays.
So I decided to fix it by using this (in the implementation not in the interface):
https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1819?view=vs-2019#change-the-property-to-a-method